Flatten JSON & TS data model Pattern element (#585)
unicode-org/message-format-wg: spec/data-model/README.md

## Patterns

> Each `Pattern` represents a linear sequence, without selectors.

Reviewer: You might want to adjust this sentence also. It might a relic of earlier times, before the addition of pattern delimiters. Perhaps:

```suggestion
Each `Pattern` is a standalone linear sequence of text and placeholders
that represents the output of a message.
```

> Each element of the `Pattern` MUST either be a non-empty string or an `Expression` object.

Reviewer: I think this is a good change, especially since `body` rhymes with `complex-body`, which means something entirely else.

> String values represent literal _text_,
> while `Expression` wraps each of the potential _expression_ shapes.
> The `Pattern` strings are the "cooked" _text_ values, i.e. escape sequences are processed.

Reviewer: The whole "cooking" thing seems just a bit informal to me. Here we refer to "`Pattern` strings" and I think that points to the problem. Perhaps:

```suggestion
String values represent literal _text_.
String values include all processing of the underlying _text_ values, that is, after escape sequences have been processed.

`Expression` values wrap each of the potential _expression_ shapes.

```

Author: Added this with a few tweaks.
